crates/core/src/host/wasmtime/wasm_instance_env.rs

```diff
@@ -169,16 +179,14 @@ impl WasmInstanceEnv {
     /// Some database errors are logged but are otherwise regarded as `Ok(_)`.
     /// See `err_to_errno` for a list.
     fn cvt(
-        mut caller: FunctionEnvMut<'_, Self>,
-        func: &'static str,
-        call: AbiCall,
-        f: impl FnOnce(FunctionEnvMut<'_, Self>, &Mem) -> WasmResult<()>,
-    ) -> RtResult<u16> {
-        let span_start = span::CallSpanStart::new(call);
+        mut caller: Caller<'_, Self>,
+        func: AbiCall,
+        f: impl FnOnce(&mut Caller<'_, Self>) -> WasmResult<()>,
+    ) -> RtResult<u32> {
```

Comment:
re the u16 -> u32 change throughout this file: is this ultimately stemming from a wasmtime api? like does it only let you have fn types whose return type impls some trait goop, and it bottoms out in only u32s etc because wasm only has i32s and not i16s?

I'm assuming the same for the anyhow::Result change in RtResult.

Reply:
Yeah, which was the same for wasmer and the difference is that wasmtime only lets you use `{i,u,f}{32,64}` for those, while wasmer was happy to truncate/zero-extend with other integer types.